test/karpenter.test.ts

```diff
@@ -114,42 +114,81 @@ describe('Unit tests for Karpenter addon', () => {
-      .version("auto")
-      .account("123567891")
-      .region("us-west-1")
-      .addOns(
-        new blueprints.KarpenterAddOn({
-          version: 'v0.29.2',
-          values: {
-            settings: {
-              aws: {
-                enableENILimitedPodDensity: true,
-                interruptionQueueName: "override-queue-name",
-              },
+
+    test("Stack creation succeeds with custom values overrides", async () => {
+        const app = new cdk.App();
+
+        const blueprint = blueprints.EksBlueprint.builder();
+
+        const stack = await blueprint
+        .version("auto")
+        .account("123567891")
+        .region("us-west-1")
+        .addOns(
+            new blueprints.KarpenterAddOn({
+            version: 'v0.29.2',
+            values: {
+                settings: {
+                aws: {
+                    enableENILimitedPodDensity: true,
+                    interruptionQueueName: "override-queue-name",
+                },
+                },
             },
-          },
-        })
-      )
-      .buildAsync(app, "stack-with-values-overrides");
-
-    const template = Template.fromStack(stack);
-    template.hasResourceProperties("Custom::AWSCDK-EKS-HelmChart", {
-      Chart: "karpenter",
+            })
+        )
+        .buildAsync(app, "stack-with-values-overrides");
+
+        const template = Template.fromStack(stack);
+        template.hasResourceProperties("Custom::AWSCDK-EKS-HelmChart", {
+        Chart: "karpenter",
+        });
+        const karpenter = template.findResources("Custom::AWSCDK-EKS-HelmChart");
+        const properties = Object.values(karpenter).pop();
+        const values = properties?.Properties?.Values;
+        expect(values).toBeDefined();
+        const valuesStr = JSON.stringify(values);
+        expect(valuesStr).toContain("defaultInstanceProfile");
+        expect(valuesStr).toContain("override-queue-name");
+        expect(valuesStr).toContain("enableENILimitedPodDensity");
     });
-    const karpenter = template.findResources("Custom::AWSCDK-EKS-HelmChart");
-    const properties = Object.values(karpenter).pop();
-    const values = properties?.Properties?.Values;
-    expect(values).toBeDefined();
-    const valuesStr = JSON.stringify(values);
-    expect(valuesStr).toContain("defaultInstanceProfile");
-    expect(valuesStr).toContain("override-queue-name");
-    expect(valuesStr).toContain("enableENILimitedPodDensity");
-  });
+
+    test("Stack creation succeeds with custom values overrides for blockDeviceMapping", async () => {
+        const app = new cdk.App();
+    
+        const blueprint = blueprints.EksBlueprint.builder();
+    
+        const stack = await blueprint
+          .version("auto")
+          .account("123567891")
+          .region("us-west-1")
+          .addOns(
+            new blueprints.KarpenterAddOn({
+              version: 'v0.29.2',
+              values: {
+                blockDeviceMappings: [{
```

Comment:
I think this incorrectly setting the helm chart values instead of the addon blockDeviceMappings property in the test.

Reply:
I've fixed this now.